File: hw/paucar/u3/hw37StrategyPattern/hw37StrategyPattern/src/main/java/ec/edu/espe/sortapp/controller/SortingContext.java
package ec.edu.espe.sortapp.controller;

import ec.edu.espe.sortapp.model.BubbleSort;
import ec.edu.espe.sortapp.model.InsertionSort;
import ec.edu.espe.sortapp.model.QuickSort;
import ec.edu.espe.sortapp.model.SortingStrategy;

/**
 *
 * @author Dennis Paucar
 */
public class SortingContext {
    private SortingStrategy strategy;
    private String strategyName;

    public void setSortStrategy(int[] array) {
        if (array.length < 30) {
            this.strategy = new BubbleSort();
            this.strategyName = "Bubble Sort";
        } else if (array.length < 100) {
            this.strategy = new InsertionSort();
            this.strategyName = "Insertion Sort";
        } else {
            this.strategy = new QuickSort();
            this.strategyName = "Quick Sort";
        }
    }

    public void sortArray(int[] array) {
        System.out.println("Unsorted array: " + java.util.Arrays.toString(array));
        setSortStrategy(array); 
        if (strategy != null) {
            strategy.sort(array);
            System.out.println("Sorted array: " + java.util.Arrays.toString(array));
            System.out.println("Sorting strategy used: " + strategyName);
        } else {
            System.out.println("No sorting strategy set.");
        }
    }
}

File: EXAM/quiroga/u3/QuirogaExamU3/StrategyPattern/pooCrudjs/public/controller/SortingContext.js

import BubbleSort from "../model/BubbleSort.js";
import InsertionSort from "../model/InsertionSort.js";
import  QuickSort from "../model/QuickSort.js";

export default class SortingContext {
    constructor() {
        this.ss = null; 
    }

    sort(data) {
        const size = data.length;
        const {sorting, sortAlgorithm} = this.setSortStrategy(size);
        const arraySorted = sorting.sort(data);
        return {arraySorted, sortAlgorithm };
    }

    setSortStrategy(size) {
        if (size > 2 && size < 5) {
            console.log("Using Bubble sort");
            return {sorting:new BubbleSort(), sortAlgorithm: "BubbleSort"};
        } else if (size >= 6 && size < 10) {
            console.log("Using Insertion sort");
            return {sorting:new InsertionSort(), sortAlgorithm: "InsertionSort"};
        } else {
            console.log("Using Quick sort");
            return {sorting:new QuickSort(), sortAlgorithm: "QuickSort"};
        }
    }
}
